services/github/github-checks-status.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('joi')
+const { NotFound, InvalidParameter } = require('..')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  state: Joi.equal('failure', 'pending', 'success'),
+}).required()
+
+module.exports = class GithubChecksStatus extends GithubAuthV3Service {
+  static category = 'build'
+  static route = {
+    base: 'github/checks-status',
+    pattern: ':user/:repo/:ref',
+  }
+
+  static examples = [
+    {
+      title: 'GitHub branch checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: 'master',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
```

Comment:
helper can abe used here
```suggestion
      staticPreview: renderBuildStatusBadge({
        status: 'success',
      }),
```